pkgs/development/compilers/cudatoolkit/flags.nix

```diff
@@ -102,38 +88,79 @@ let
     "-gencode=arch=compute_${dropDot computeCapability},code=${feat}_${dropDot computeCapability}"
   );
 
-  # cudaRealArches :: List String
-  # The real architectures are physical architectures supported by the CUDA version.
-  # For example, "sm_80".
-  cudaRealArches = archMapper "sm" cudaComputeCapabilities;
-
-  # cudaVirtualArches :: List String
-  # The virtual architectures are typically used for forward compatibility, when trying to support
-  # an architecture newer than the CUDA version allows.
-  # For example, "compute_80".
-  cudaVirtualArches = archMapper "compute" cudaComputeCapabilities;
-
-  # cudaArches :: List String
-  # By default, build for all supported architectures and forward compatibility via a virtual
-  # architecture for the newest supported architecture.
-  cudaArches = cudaRealArches ++
-    lists.optional (config.cudaForwardCompat or true) (lists.last cudaVirtualArches);
-
-  # cudaGencode :: List String
-  # A list of CUDA gencode arguments to pass to NVCC.
-  cudaGencode =
-    let
-      base = gencodeMapper "sm" cudaComputeCapabilities;
-      forwardCompat = gencodeMapper "compute" [ (lists.last cudaComputeCapabilities) ];
-    in
-    base ++ lists.optionals (config.cudaForwardCompat or true) forwardCompat;
+  formatCapabilities = { cudaCapabilities, enableForwardCompat ? true }: rec {
```

Comment:
@ConnorBaker I needed the names of "currently enabled" cuda platforms (now exposed in` archNames`), so I sort of moved things around a little and ended up cutting names after I noticed myself writing things like `cudaCapabilities ? cudaPackages.cudaFlags.cudaArchNames` (something too many cucucucudas). I think we might want to go even more radical (e.g. why `cudaFlags.cudaCapabilities` and not `cudaFlags.capabilities`). Open to any suggestions and recommendations here

Reply:
You're right, there's a lot of repetition! I just saw that inside `cudaPackages` things generally had a `cuda` prefix and made sure to do that for everything inside `cudaFlags`. I'm down to change to something else if you think it'd be more helpful -- `cudaPackages.flags.capabilities` is certainly less a mouthful.

Reply:
I'm OK with `cudaFlags`, because we often `inherit (cudaPackages) cudaFlags` and use it as `cudaFlags` ever since